internal/pkg/service/buffer/worker/service/slice_watch.go

```diff
@@ -73,9 +73,10 @@ func NewActiveSlicesWatcher(ctx context.Context, wg *sync.WaitGroup, logger log.
 			// Trigger change listeners
 			if len(events) > 0 {
 				w.lock.Lock()
-				for _, l := range w.listeners {
+				for id, l := range w.listeners {
 					if w.countSlices(l.fileKey) == 0 {
 						close(l.ch)
+						delete(w.listeners, id)
 					}
```

Comment:
Tu mohla nastat situacia, ze sme dany chanel chceli zavriet viac krat, chybalo tu odstranenie `listener`.

Reply:
Riesil to aj `defar stop()` na riadku `102` nizsie, ale aj tak sa mohlo stat, ze sa tento kod zavolal viac krat, kym sa vykonalo `defer stop()`.

Reply:
`delete` nad map je mozne zavolat, aj ked ten kluc uz neexistuje + pristup k map je chraneny zamkom.